Improved build script

DeclanHoare opened the pull request:
Changed a few aspects of the build script:
* Running with "bash -e", so will quit on errors
* $CC now has a default of ~~"cc", which is usually a symlink to the system's default C compiler~~ gcc; it can also be overridden from the environment
* Use sudo if available to only run the mount-related commands as root
* "mnt" is removed using rmdir rather than rm -rf, because it should be empty by that point
* \_\_DANKOS\_\_ is defined in C programs for conditional target checks

Commit:
Improved build script

Changed a few aspects of the build script:
* Running with "bash -e", so will quit on errors
* $CC now has a default of "cc", which is a symlink to the system's default C compiler
* Use sudo if available to only run the mount-related commands as root

Commit:
Added __DANKOS__ define

so you can use #ifdef's in your C programs to detect DankOS as a
target platform.

Commit:
$CC can be set from environment

$CC is now set to ${CC:-cc}, meaning you can override it by running
CC=clang ./make_me.sh, but it will default to "cc" if the variable
is not already set in your environment.

wubbalubbadubdubonline commented:
i like the bash -e change, however, the change of the c compiler is a no-go since DankOS uses gnu extended inline assembly and is therefore dependent on GCC to compile.

wubbalubbadubdubonline commented:
furthermore it seems you accidentally placed a single quote after one of your semicolons in the edit

RogueAI42 commented:
Give me a sec, I'll fix the CC issue. The single quote is correct.

Commit:
use gcc

gcc is the only (?) compiler capable of building DankOS so I've made it
the default.

RogueAI42 commented:
Alright, I've fixed that. Whoops.

wubbalubbadubdubonline commented:
looks good, builds fine, gj

wubbalubbadubdubonline commented:
i don't think i have merge permissions though

Mintz98 commented:
alright, im merging it, thanks
